opentelemetry/src/trace/context.rs

```diff
@@ -55,7 +55,9 @@ impl SpanRef<'_> {
         if let Some(ref inner) = self.0.inner {
             match inner.lock() {
                 Ok(mut locked) => f(&mut locked),
-                Err(err) => global::handle_error(err),
+                Err(_err) => {
+                    otel_error!(name: "SpanRef.with_inner_mut", otel_name = "SpanRef.with_inner_mut", error = format!("{:?}", _err));
```

Comment:
instead of repeating otel_name, lets do it in the wrapper itself for now, until `tracing:fmt` is fixed to display event name.

Reply:
yes, I thought so, but then this will force this behavior for anyone using these macros in their custom exporters or processors.  Thought not a big reason for not adding it in macros :)